pkg/provider/azure_loadbalancer_backendpool.go

```diff
@@ -614,8 +618,11 @@ func (bi *backendPoolTypeNodeIP) ReconcileBackendPools(clusterName string, servi
 		klog.V(2).Infof("bi.ReconcileBackendPools for service (%s) and vmSet (%s): ensuring the LB is decoupled from the VMSet", serviceName, vmSetName)
 		shouldRefreshLB, err = bi.VMSet.EnsureBackendPoolDeleted(service, lbBackendPoolIDsSlice, vmSetName, lb.BackendAddressPools, true)
 		if err != nil {
-			klog.Errorf("bi.ReconcileBackendPools for service (%s): failed to EnsureBackendPoolDeleted: %s", serviceName, err.Error())
-			return false, false, false, err
+			if !retry.HasStatusForbiddenOrIgnoredError(err) {
```

Comment:
ensureBackendPoolDeleted decouples vmss and vm from backend pool instead of deleting the backend pool. I am not sure if it will return 404 when updating an non-existent vmss/vmssvm.

Reply:
I think it will return 404 when it vmssCache.Get() with non-existing vmss.
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/5d65ed33959ebe6a98bdc0aabaf8c9d3422c0405/pkg/provider/azure_vmss.go#L1645